tests/test_memory.py

```diff
@@ -279,3 +280,28 @@ def test_memory_buffer_bytearray():
     assert len(byte_array) == 1114112
     assert byte_array[0:3] == b'\x01\x02\x03'
     assert byte_array[3:9].decode() == 'Wasmer'
+
+def test_memory_buffer_supports_ctypes():
+    c_uint8_4 = ctypes.c_uint8 * 4
+
+    memory = instance().exports.memory
+
+    arr = c_uint8_4.from_buffer(memory.buffer)
+    arr[0] = 0b00000001
+    arr[1] = 0b00000100
+    arr[2] = 0b00010000
+    arr[3] = 0b01000000
+
+    byte_array = bytearray(memory.buffer)
+    assert byte_array[0] == 0b00000001
+    assert byte_array[1] == 0b00000100
+    assert byte_array[2] == 0b00010000
+    assert byte_array[3] == 0b01000000
+
+def test_memory_buffer_supports_keeps_object_alive():
+    # IMPL NOTE: make sure the memory instance is immediately del'd
```

Comment:
To be sure the memory instance is deleted, you have to either you `del` (but it's not sure the GC will collect it) or (better) force the GC to collect it.

Reply:
You're right. I'll add the `gc.collect` part. I think CPython will always collect eagerly, if there are no cycles (as is the case here), but probably this does not apply to other interpreters.

The `del` call is not required, as the `memory.buffer` is only used as a temporary inside the line. But I guess making it two lines and explicitly deteting the ref to buffer is more self-documenting.

Reply:
> The del call is not required, as the memory.buffer is only used as a temporary inside the line. But I guess making it two lines and explicitly deteting the ref to buffer is more self-documenting.

Exactly, that was my idea. Thanks for reading in my minds :-p.